### ISSUE TITLE
Agent doesn't take X-Forwarded-For header into account

### ISSUE THREAD

**Issue by ashevtsov-wawa** [aws/aws-xray-java-agent#78]
We keep seeing `client_ip` field set as `127.0.0.1` even though `X-Forwarded-For` header is set in the request. 
From [X-Ray Concepts](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html):
> If a load balancer or other intermediary forwards a request to your application, X-Ray takes the client IP from the X-Forwarded-For header in the request instead of from the source IP in the IP packet.

I wonder if it's due to the use of `getLocalIPAddress()` method [here](https://github.com/aws/aws-xray-java-agent/blob/00d57b187b22660cb37184385239eef912c06f3f/aws-xray-agent/src/main/java/com/amazonaws/xray/agent/runtime/handlers/upstream/ServletHandler.java#L115)
```java
    private void populateHeaderToTransactionState(HttpNetworkProtocolRequestEvent requestEvent, XRayTransactionState transactionState) {
        transactionState.withHost(requestEvent.getHost())
                .withMethod(requestEvent.getMethod())
                .withUrl(requestEvent.getURL())
                .withUserAgent(requestEvent.getUserAgent())
                .withClientIP(requestEvent.getLocalIPAddress())
                .withTraceheaderString(requestEvent.getHeaderData(HEADER_KEY));
```

**Comment by willarmiros** [comment]
Hi @ashevtsov-wawa,

Thanks for raising this issue! You're absolutely correct that this is a missing feature on the agent. I've opened #79 to address this for now.

**Comment by ashevtsov-wawa** [comment]
@willarmiros thanks for looking into this issue and for raising the PR. I left one comment.

**Comment by ashevtsov-wawa** [comment]
@willarmiros I saw that the PR got merged a month ago. Wondering when a new version is going to be released?

**Comment by willarmiros** [comment]
@ashevtsov-wawa apologies I was out on vacation. We can't give exact release dates but will be working to get a release out as soon as possible!

**Comment by willarmiros** [comment]
Fixed in 2.9.0
